contracts/multisig/src/key.rs

```diff
@@ -1,55 +1,113 @@
+use crate::{secp256k1::ecdsa_verify, types::MsgToSign, ContractError};
 use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{HexBinary, StdError, StdResult};
 use cw_storage_plus::{KeyDeserialize, PrimaryKey};
+use enum_display_derive::Display;
 use serde::{de::Error, Deserializer};
-use std::marker::PhantomData;
-
-use crate::{secp256k1::ecdsa_verify, types::MsgToSign, ContractError};
+use std::fmt::Display;
 
 #[cw_serde]
-#[derive(Copy)]
+#[derive(Copy, Display)]
 pub enum KeyType {
     Ecdsa,
 }
 
 #[cw_serde]
 #[derive(PartialOrd, Ord, Eq)]
-pub enum Signature<T>
-where
-    T: Recoverability,
-{
-    Ecdsa(HexBinary, PhantomData<T>),
+pub enum Signature {
+    Ecdsa(NonRecoverable),
+    EcdsaRecoverable(Recoverable),
+}
+
+#[cw_serde]
+#[derive(Ord, PartialOrd, Eq)]
+pub struct NonRecoverable(HexBinary);
+
+impl NonRecoverable {
+    const LEN: usize = 64;
+
+    pub fn to_recoverable(
+        &self,
+        msg: &[u8],
+        pub_key: &PublicKey,
+        recovery_transform: impl FnOnce(u8) -> u8,
+    ) -> Result<Recoverable, ContractError> {
```

Comment:
```suggestion
    pub fn to_recoverable(
        &self,
        msg: &[u8],
        pub_key: &PublicKey,
        recovery_transform: impl FnOnce(u8) -> u8,
    ) -> Result<HexBinary, ContractError> {
```

just return `HexBinary` and make it a free function, instead of a method